test/functional/api/buffer_updates_spec.lua

```diff
@@ -741,3 +742,87 @@ describe('API: buffer events:', function()
   end)
 
 end)
+
+describe('API: buffer events:', function()
+  before_each(function()
+    helpers.clear()
+  end)
+
+  local function lines_subset(first, second)
+    for i = 1,#first do
+      if first[i] ~= second[i] then
+        return false
+      end
+    end
+    return true
+  end
+
+  local function lines_equal(f, s)
+    return lines_subset(f, s) and lines_subset(s, f)
+  end
+
+  local function did_match_somwhere(buffer_lines, expected_lines)
+    local msg = next_msg()
+
+    while(msg ~= nil) do
+      local event = msg[2]
+      if event == "nvim_buf_lines_event" then
+        local args = msg[3]
+        local starts = args[3]
+        local newlines = args[5]
+
+        -- Size of the contained neovim instance is 23 lines, this might change
+        -- with the test setup. Note updates are continguous.
+        assert(#newlines <= 23)
+
+        for i = 1,#newlines do
+          buffer_lines[starts + i] = newlines[i]
+        end
+        -- we don't compare the msg area of the embedded nvim, it's too flakey
+        buffer_lines[23] = nil
+
+        if lines_equal(buffer_lines, expected_lines) then
+          return true
+        end
+      end
+      msg = next_msg()
+    end
+    return false
+  end
+
+  it('terminal with a nested nvim instance', function()
+    local buffer_lines = {}
+    local expected_lines = {}
+    command("terminal " .. nvim_dir ..'/nvim -n -c "set shortmess+=A"')
```

Comment:
`-u NONE -i NONE` might be a good idea as well, else a developer's local config may get in the way.